src/popsift/common/sync_queue.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+
+namespace popsift {
+
+/*************************************************************
+ * SyncQueue
+ * This is a basic alternative to the Boost sync_queue class.
+ * It lets threads push and pull items off a queue in a thread
+ * safe manner.
+ *************************************************************/
+template<typename T>
+class SyncQueue {
+ public:
+  SyncQueue() = default;
+
+  /* Push an item onto the queue and signal it's available. */
```

Comment:
please use doxygen format (same for the others)
```suggestion
  /**
   * @brief Push an item onto the queue and signal it is available.
   * @param[in] value the item to add to the queue.
   */
```

Reply:
I've added doxygen comments to the class and its functions, but I'm not especially fluent in doxygen's comment style (i.e. it could use a second review).

Reply:
perfect!